app/models/portal/teacher.rb

```diff
@@ -159,6 +162,15 @@ def my_classes_url(protocol, host)
     Rails.application.routes.url_helpers.api_v1_classes_mine(protocol: protocol, host: host)
   end
 
+  def add_recent_collection_page(project)
```

Comment:
Since this using the model terminology, I'd suggest the name of this method should match the model fields.
So perhaps `record_project_view`